src/wp-admin/includes/file.php

```diff
@@ -2555,3 +2555,41 @@ function wp_opcache_invalidate( $filepath, $force = false ) {
 
 	return false;
 }
+
+/**
+ * Wrapper for PHP filesize with filters and casting the result as an integer.
+ *
+ * @since 6.0.0
+ *
+ * @link https://www.php.net/manual/en/function.filesize.php
+ *
+ * @param string $path Path to the file.
+ * @return int The size of the file in bytes, or 0 in the event of an error.
+ */
+function wp_filesize( $path ) {
```

Comment:
This is awesome to have. However, note that there are a ton more `filesize()` usages in core than are being updated here. Many of those other usages are not image-related though.

Do you intend to replace all WP core calls to `filesize()` with a call to this function separately? If yes, sgtm. If not, we may need to re-focus this function to be specific to images. Curious to hear your thinking about this.

Reply:
There is a discussion on the ticket about caching these calls.

Using transients was discouraged because it can lead to database writes on front-end requests. However this could use `wp_cache_*()` to provide some benefit if the filesize is requested multiple times in a single request/to sites with a persistent cache.

The cache would need to be deleted when updated.

Reply:
You are ahead of me here. My plans is use this function in more places in core. Anytime a file is stored in uploads or temp directory. I was going to make a ticket once this was merged.

Reply:
@peterwilsoncc I don't think storing these values in caches adds a lot of value. The idea here to the file stored is stored forever in post meta. Storing it is object cache, is just double caching it. It would also fill up an object cache pointless. If you really really store the value in object cache, then you can, there are two filter here to enable you to do this, however, I am not sure I see the value.